torch/nn/utils/clip_grad.py

```diff
@@ -15,7 +17,7 @@ def clip_grad_norm(parameters, max_norm, norm_type=2):
     Returns:
         Total norm of the parameters (viewed as a single vector).
     """
-    parameters = list(filter(lambda p: p.grad is not None, parameters))
+    parameters = filter(lambda p: p.grad is not None, parameters)
```

Comment:
This function is different. It needs the `list`, because it's going to iterate over the parameters multiple times. It wasn't necessary in your case, because you do it only once. Can you please revert this?

Reply:
Sorry about that!